File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> bucket;
        
        for(int i=0;i<s.size();i++)
        {
            if(bucket.empty())
            {
                bucket.push(s[i]);
            }
            else if( (bucket.top()=='(' && s[i]==')') || (bucket.top()=='[' && s[i]==']') || (bucket.top()=='{' && s[i]=='}'))
            {
                bucket.pop();
            }
            else
            {
                bucket.push(s[i]);
            }
        }
        return bucket.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> bucket = new Stack<>();
        for(int i=0;i<s.length();i++){
            if(bucket.isEmpty()){
                bucket.add(s.charAt(i));
            }
            else if((bucket.peek()=='(' && s.charAt(i)==')') || (bucket.peek()=='{' && s.charAt(i)=='}') || (bucket.peek()=='[' && s.charAt(i)==']')){
                bucket.pop();
            }
            else{
                bucket.add(s.charAt(i));
            }
        }
        return bucket.isEmpty();
    }
}